dev-libs/leveldb/leveldb-1.20.recipe

```diff
@@ -0,0 +1,81 @@
+SUMMARY="A fast key-value storage library written at Google"
+DESCRIPTION="LevelDB stores keys and values in arbitrary byte arrays, and \
+data is sorted by key. It supports batching writes, forward and backward \
+iteration, and compression of the data via Google's Snappy compression \
+library.
+LevelDB is not an SQL database. Like other NoSQL and Dbm stores, it does \
+not have a relational data model and it does not support SQL queries. Also, \
+it has no support for indexes. Applications use LevelDB as a library, as it \
+does not provide a server or command-line interface."
+HOMEPAGE="http://leveldb.org/"
+COPYRIGHT="2017 The LevelDB Authors"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/google/leveldb/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="f5abe8b5b209c2f36560b75f32ce61412f39a2922f7045ae764a2c23335b6664"
+SOURCE_FILENAME="leveldb-$portVersion.tar.gz"
+PATCHES="leveldb-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="1"
```

Comment:
Shouldn't this be `libVersion="1.20"` (or, even better,  `libVersion="$portVersion"`) ?
Yeah, I know I already commented about this.
BTW, I just added another comment [here](https://github.com/haikuports/haikuports/pull/2555#discussion_r188390049).

Reply:
HP packages the libleveldb.so.1.20 into the develop package, so the debuginfo part cannot find it as it doesn't search in the packaging/develop subfolder.
No idea if it is ok or not.

Reply:
Hmm, I think the recipe is missing a call to `fixPkgconfig` after the call to `prepareInstalledDevelLibs`.

Reply:
Sadly that doesn't help.

Reply:
It should be `libVersion="$portVersion"` IMO, based on [upstream](https://github.com/google/leveldb/blob/v1.20/Makefile#L124)

> HP packages the libleveldb.so.1.20 into the develop package, so the debuginfo part cannot find it as it doesn't search in the packaging/develop subfolder.
No idea if it is ok or not.

According to the [`Makefile`](https://github.com/google/leveldb/blob/v1.20/Makefile#L130), the `libleveldb.so.1` and `libleveldb.so` were symlinked to `libleveldb.so.1.20`. This means that the package could not be used without the `_devel` version installed, so not OK from me. That also means `prepareInstalledDevelLibs` might need to be fixed up.

Reply:
Indeed not ok, and this is exacly what prepareInstalledDevelLibs is supposed to handle. In libs there should be only the .1.20 file, and in develop/lib there should be 3 symlinks to it: .so, .so.1 and .so.1.20.

The recipe should install everything in lib and let prepareInstalledDevelLibs handle this. So how did we end with an actual .so.1.20 file in developLibs in the first place?

Reply:
It is magic, better to say: idk. I can think that the compat string is wrong and HP picks the wrong file or symlink, but i'm not sure how does it works under the hood.

Reply:
The fix is simple: [pass `--no-dereference`](#r188610794) to `cp`  when copying the shared lib and its symlinks.

Reply:
Oh, snap!